src/ui/components/NetworkMonitor/index.tsx

```diff
@@ -111,28 +112,18 @@ export const NetworkMonitor = ({
                     dispatch(fetchRequestBody(row.id, row.point.point));
                   }
 
-                  setSelectedRequest(row);
+                  dispatch(showRequestDetails(row.id));
                 }}
                 seek={seek}
-                selectedRequest={selectedRequest}
+                selectedRequest={data.find(request => request.id === selectedRequestId)}
               />
             }
             endPanel={
-              selectedRequest ? (
-                loadedRegions &&
-                getPointIsInLoadedRegion(loadedRegions, selectedRequest.point.point) ? (
-                  <RequestDetails
-                    closePanel={closePanel}
-                    cx={cx}
-                    request={selectedRequest}
-                    responseBody={responseBodies[selectedRequest.id]}
-                    requestBody={requestBodies[selectedRequest.id]}
-                    frames={frames[selectedRequest?.point.point]}
-                    selectFrame={selectFrame}
-                  />
-                ) : (
-                  <RequestDetailsUnavailable closePanel={closePanel} />
-                )
+              selectedRequestId ? (
+                <RequestDetails
+                  cx={cx}
+                  request={data.find(request => request.id === selectedRequestId)!}
```

Comment:
What do you think about making these things selectors as well rather than passing them in? If it's difficult don't worry about it, but it seems like since RequestDetails gets 90% of its data straight from the store it's weird to have it rely on a couple of things that come from the store through its parent 🤷